### PR TITLE
Initial release of terraform-aws-dms-iam-role module

### PR DESCRIPTION
**Pull request by nin9swells**
AWS DMS requires us to create role before using it.
This module will create two roles by default, `dms-vpc-role` and `dms-cloudwatch-logs-role`.
If user use Amazon Redshift as a target database, user must create `dms-access-for-endpoint` IAM role in their AWS account.

Source: https://docs.aws.amazon.com/dms/latest/userguide/CHAP_Security.APIRole.html

### CODE REVIEW

**Comment by rafikurnia** [comment]
Hi @nin9swells, I am wondering, do we actually need a dedicated module for those resources. How if we put them in [traveloka/terraform-aws-common-iam-roles](https://github.com/traveloka/terraform-aws-common-iam-roles) module instead, what do you think?

**Comment by nin9swells** [comment]
@rafikurnia,
We had offline discussion and we have this thing cleared up. I will summarize it in this comment.

This module is meant to simplify IAM role creation, that required for DMS. Since it's only useful for DMS, I don't think this belongs to common IAM roles module.

File: data.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+data "aws_iam_policy_document" "assume_role_policy_for_dms_and_redshift" {
+  statement = {
+    actions = ["sts:AssumeRole"]
+
+    principals = {
+      type = "Service"
+
+      identifiers = [
+        "dms.amazonaws.com",
+        "redshift.amazonaws.com",
```
Comment (rafikurnia) [inline data.tf]: Just want to ask, is the redshift part required?

Comment (nin9swells) [reply data.tf]: It's required based on AWS documentation

**Pull request merged**